pkgs/tools/misc/fastfetch/default.nix

```
@@ -90,6 +92,16 @@ stdenv.mkDerivation (finalAttrs: {
   cmakeFlags = [
     "-DCMAKE_INSTALL_SYSCONFDIR=${placeholder "out"}/etc"
     "-DENABLE_SYSTEM_YYJSON=YES"
+  ] ++ lib.optionals (!x11Support) [
+    "-DENABLE_GLX=NO"
+    "-DENABLE_VULKAN=NO"
+    "-DENABLE_WAYLAND=NO"
+    "-DENABLE_X11=NO"
+    "-DENABLE_XCB=NO"
+    "-DENABLE_XCB_RANDR=NO"
+    "-DENABLE_XFCONF=NO"
+    "-DENABLE_XFCONF=NO"
+    "-DENABLE_XRANDR=NO"
```

Since we are here...

```suggestion
    (lib.cmakeBool "ENABLE_GLX" x11Support)
    (lib.cmakeBool "ENABLE_VULKAN" x11Support)
    (lib.cmakeBool "ENABLE_WAYLAND" x11Support)
    (lib.cmakeBool "ENABLE_X11" x11Support)
    (lib.cmakeBool "ENABLE_XCB" x11Support)
    (lib.cmakeBool "ENABLE_XCB_RANDR" x11Support)
    (lib.cmakeBool "ENABLE_XFCONF" x11Support)
    (lib.cmakeBool "ENABLE_XRANDR" x11Support)
```

P.S.: why xfconf appeared two times?
P.P.S.: why Wayland support depends on X11?

---

Thanks, fixed.

> P.S.: why xfconf appeared two times?

Typo.

> P.P.S.: why Wayland support depends on X11?

Fixed.